posthog/api/survey.py

```diff
@@ -202,6 +202,9 @@ def update(self, instance: Survey, validated_data):
                 validated_data["targeting_flag_id"] = None
             validated_data.pop("remove_targeting_flag")
 
+        # make sure instance.targeting_flag is gone
+        instance.refresh_from_db()
```

Comment:
opted this route because it makes it such that we don't have to remember this and can start with a clean slate for all the rest of the update logic (the key hanging around can lead to issues later on for example, if we pass it into a serializer that doesn't understand the key)